packages/scene-composer/src/components/three-fiber/AnimationComponent/__tests__/AnimationComponent.spec.tsx

```diff
@@ -0,0 +1,209 @@
+  }));
+  const Mixer = new THREE.AnimationMixer({ animations: ['Action.013'] } as any);
+
+  jest.mock('three', () => ({
+    findByName: jest.fn(),
+    clipAction: jest.fn(),
+  }));
+
+  expect(toggleIsAnimationPaused(Mixer, true, ['Action.013'])).toBe(undefined);
+});
+
+const mockNode = {
+  ref: 'ce7802dd-477e-4bae-b787-374aea3fce77',
+  name: 'AIMIMATED_MIXER',
+  transform: {
+    position: [-0.2548017433916856, 0.1565180250896893, 1.9222755178165505],
+    rotation: [0, 0, 0],
+    scale: [1, 1, 1],
+  },
+  transformConstraint: {},
+  properties: {},
+  components: [
+    {
+      ref: 'f7bdb4b0-8627-4c83-825e-bace9874bc08',
+      type: 'ModelRef',
+      uri: 'AIMIMATED_MIXER.gltf',
+      modelType: 'GLTF',
+    },
+    {
+      selector: 1,
+      ref: 'a2159735-991c-41da-a217-c39445a47b46',
+      currentAnimations: ['Action.018'],
+      uri: 'AIMIMATED_MIXER.gltf',
+      type: 'Animation',
+    },
+  ],
+  childRefs: [],
+} as unknown as ISceneNodeInternal;
+
+const object3D = new THREE.Object3D();
+object3D.add(new THREE.Scene());
+const baseState = {
+  setScene: jest.fn().mockReturnValue(mockNode),
+  getObject3DBySceneNodeRef: jest.fn().mockReturnValue(object3D),
+  getObjectByName: jest.fn().mockReturnValue(new THREE.Scene()),
+  document: {
+    nodeMap: {
+      'test-uuid': mockNode,
+    },
+  },
+};
+
+describe('<AnimationComponent />', () => {
+  const effects = [];
+  beforeEach(() => {
+    jest.clearAllMocks();
+    (useEffect as jest.Mock).mockImplementation((effect, dep) => {
+      effects.push(effect);
+      return effect;
+    });
+  });
+
+  it('should test a simple animation component', () => {
+    const mockNode = {
+      ref: 'ce7802dd-477e-4bae-b787-374aea3fce77',
+      name: 'AIMIMATED_MIXER',
+      Scene: { s: 'Action.013' },
+      transform: {
+        position: [-0.2548017433916856, 0.1565180250896893, 1.9222755178165505],
+        rotation: [0, 0, 0],
+        scale: [1, 1, 1],
+      },
+      transformConstraint: {},
+      properties: {},
+      components: [
+        {
+          ref: 'f7bdb4b0-8627-4c83-825e-bace9874bc08',
+          type: 'ModelRef',
+          uri: 'AIMIMATED_MIXER.gltf',
+          modelType: 'GLTF',
+        },
+        {
+          selector: 1,
+          ref: 'a2159735-991c-41da-a217-c39445a47b46',
+          currentAnimations: ['Action.018'],
+          uri: 'AIMIMATED_MIXER.gltf',
+          type: 'Animation',
+        },
+      ],
+      childRefs: [],
+    } as unknown as ISceneNodeInternal;
+
+    const mockComponent: IAnimationComponentInternal = {
+      ref: generateUUID(),
+      uri: 'Animated_Mixer.gltf',
+      currentAnimations: ['Action.018'],
+    } as IAnimationComponentInternal;
+
+    useStore('default').setState(baseState);
+    const { container } = render(<AnimationComponent node={mockNode} component={mockComponent as IComponent} />);
+
+    effects[0]({ s: 'Action.014' }, false);
+    effects[1]({ s: 'Action.014' }, false);
+    effects[3];
```

Comment:
like this part, super hard for people to understand what they are doing, why 0,1 and 3, etc